plugins/module_utils/vmware.py

```diff
@@ -218,3 +224,141 @@ def is_vcenter(self):
             return True
         elif api_type == 'HostAgent':
             return False
+
+    def __get_objs_by_name(self, vimtype, name, first_only=False):
+        """
+        Get any vsphere object associated with a given text name and vim type.
+        Args:
+            vimtype: The type of object to search for
+            name: The name of the object to search for
+            first_only: If true, return once the first object is found.
+                        Useful when names must be unique
+        Returns:
+            list(object) or list() if no matches are found
+        """
+        objs = []
+        container = self.content.viewManager.CreateContainerView(self.content.rootFolder, vimtype, True)
+        for c in container.view:
+            if c.name == name:
+                if first_only:
+                    return c
+                objs += [c]
+
+        return objs if objs else None
```

Comment:
```suggestion
    def __get_objs_by_name(self, vimtype, name_or_id, first_only=False):
        """
        Get any vsphere object associated with a given text name or ID and vim type.
        Args:
            vimtype: The type of object to search for
            name_or_id: The name or the ID of the object to search for
            first_only: If true, return once the first object is found.
                        Useful when names must be unique
        Returns:
            list(object) or list() if no matches are found
        """
        obj = list()
        container = self.content.viewManager.CreateContainerView(
            self.content.rootFolder, vimtype, True)

        for c in container.view:
            if name_or_id in [c.name, c._GetMoId()]:
                if return_all is False:
                    return c
                else:
                    obj.append(c)

        if len(obj) > 0:
            return obj
        else:
            # for backwards-compat
            return None
```

Reply:
that makes sense, ill make these changes!